tests/windows_test.go

```diff
@@ -0,0 +1,235 @@
+						DiskDevice: v1.DiskDevice{Disk: &v1.DiskTarget{Bus: "sata"}},
+					},
+				},
+			},
+		},
+		Volumes: []v1.Volume{
+			{
+				Name: windowsDisk,
+				VolumeSource: v1.VolumeSource{
+					Ephemeral: &v1.EphemeralVolumeSource{
+						PersistentVolumeClaim: &k8sv1.PersistentVolumeClaimVolumeSource{
+							ClaimName: tests.DiskWindows,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	tests.BeforeAll(func() {
+		tests.SkipIfNoWindowsImage(virtClient)
+	})
+
+	BeforeEach(func() {
+		tests.BeforeTestCleanup()
+		windowsVm = tests.NewRandomVM()
+		windowsVm.Spec = windowsVmSpec
+	})
+
+	It("should success to start", func() {
+		vm, err := virtClient.VM(tests.NamespaceTestDefault).Create(windowsVm)
+		Expect(err).To(BeNil())
+		tests.WaitForSuccessfulVMStartWithTimeout(vm, 180)
+	}, 300)
+
+	It("should success to stop", func() {
+		By("Starting a VM")
+		vm, err := virtClient.VM(tests.NamespaceTestDefault).Create(windowsVm)
+		Expect(err).To(BeNil())
+		tests.WaitForSuccessfulVMStartWithTimeout(vm, 180)
+
+		By("Stopping the VM")
+		err = virtClient.VM(tests.NamespaceTestDefault).Delete(vm.Name, &metav1.DeleteOptions{})
+		Expect(err).To(BeNil())
+	}, 300)
+
+	Context("with winrm connection", func() {
+		var winrmcliPod *k8sv1.Pod
+		var cli []string
+		var output string
+
+		BeforeEach(func() {
+			By("Creating winrm-cli pod for future use")
+			winrmcliPod = &k8sv1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: winrmCli + rand.String(5)},
+				Spec: k8sv1.PodSpec{
+					Containers: []k8sv1.Container{
+						{
+							Name:    winrmCli,
+							Image:   fmt.Sprintf("%s/%s:%s", tests.KubeVirtRepoPrefix, winrmCli, tests.KubeVirtVersionTag),
+							Command: []string{"sleep"},
+							Args:    []string{"3600"},
+						},
+					},
+				},
+			}
+			winrmcliPod, err = virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Create(winrmcliPod)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Starting windows VM")
+			vm, err := virtClient.VM(tests.NamespaceTestDefault).Create(windowsVm)
+			Expect(err).To(BeNil())
+			tests.WaitForSuccessfulVMStartWithTimeout(vm, 180)
+
+			vm, err = virtClient.VM(tests.NamespaceTestDefault).Get(vm.Name, metav1.GetOptions{})
+			cli = []string{
+				winrmCliCmd,
+				"-hostname",
+				vm.Status.Interfaces[0].IP,
+				"-username",
+				windowsVmUser,
+				"-password",
+				windowsVmPassword,
+			}
+		})
+
+		It("should have correct UUID", func() {
+			command := append(cli, "wmic csproduct get \"UUID\"")
+			By(fmt.Sprintf("Running \"%s\" command via winrm-cli", command))
+			Eventually(func() error {
+				output, err = tests.ExecuteCommandOnPod(
+					virtClient,
+					winrmcliPod,
+					winrmcliPod.Spec.Containers[0].Name,
+					command,
+				)
+				return err
+			}, time.Minute*5, time.Second*15).ShouldNot(HaveOccurred())
+			By("Checking that the Windows VM has expected UUID")
+			Expect(output).Should(ContainSubstring(strings.ToUpper(windowsFirmware)))
+		}, 360)
+
+		It("should have pod IP", func() {
```

Comment:
Instead of fetching the pod, you can simply wait for the vm to run and look up the ip from its status.

Reply:
fixed